lib/blemulator/internal/simulation_manager_mixins/characteristics_mixin.dart

```diff
@@ -38,7 +38,7 @@ mixin CharacteristicsMixin on SimulationManagerBase {
     SimulatedCharacteristic targetCharacteristic = targetPeripheral
         .getCharacteristicForService(serviceUuid, characteristicUUID);
 
-    _throwErrorIfNull(targetCharacteristic);
+    _throwErrorIfNull(targetCharacteristic, characteristicUUID);
```

Comment:
What if targetPeripheral is null?

Reply:
StateError from _peripherals.values.firstWhere() will be thrown and passed as a result of this "_readCharacteristicForDevice" method. There is already a task about checking if the device that we want to perform BLE operation on is known and if we are connected to it.